### PR TITLE
Bugfix: use getConfig() to get correct creds for blockstorage provider

### PR DESCRIPTION
**Pull request by SupriyaKasten**
## Change Overview

Bugfix: use getConfig() to get correct creds for blockstorage provider when calling `provider.SnapshotCreateWaitForCompletion()` func

Without this change we did not check if `role` was set in param.Profile. With this PR, `getConfig()` will take care of providing correct configuration for the blockstorage provider. 

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [x] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Issues

- #XXX

## Test Plan

- [ ] :muscle: Manual
- [x] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: pkg/function/wait_for_snapshot_completion.go [hunk 1]
```diff
@@ -65,22 +65,16 @@ func waitForSnapshotsCompletion(ctx context.Context, snapshotinfo string, profil
...
 	}
+
 	for _, pvcInfo := range PVCData {
-		config := make(map[string]string)
 		if err = ValidateLocationForBlockstorage(profile, pvcInfo.Type); err != nil {
 			return errors.Wrap(err, "Profile validation failed")
 		}
-		switch pvcInfo.Type {
-		case blockstorage.TypeEBS:
+		config := getConfig(profile, pvcInfo.Type)
```
Comment (pavannd1) [inline pkg/function/wait_for_snapshot_completion.go]: `getConfig` is not handling unsupported storage provider. Will something else handle `default` case?

Comment (SupriyaKasten) [reply pkg/function/wait_for_snapshot_completion.go]: Oh I assumed `ValidateLocationForBlockstorage ` func takes care of that. Let me update the PR.

Comment (pavannd1) [reply pkg/function/wait_for_snapshot_completion.go]: That also doesn't. It was called `ValidateProfile` before I renamed it. Do you think we should handle the check there?

Comment (SupriyaKasten) [reply pkg/function/wait_for_snapshot_completion.go]: I think so, since it will fail at validation stage, I just added a commit to this PR for the change.

**Pull request merged**